packages/common/src/utils/personIdValidatorUtils.ts

```diff
@@ -1,29 +1,72 @@
-export const isPersonIdValid = (personId: string): boolean => {
-  const cleanPersonId = personId.replace('-', '')
-  const numbers = cleanPersonId.match(/^(\d)(\d)(\d)(\d)(\d)(\d)(\d)(\d)(\d)(\d)(\d)(\d)$/)
-  let checkSum = 0
+import { intervalToDuration, isFuture, isValid, parseISO } from 'date-fns'
 
-  if (numbers == null) {
-    return false
-  }
+const MAXIMUM_AGE = 125
 
-  let n
-  for (let i = 3; i <= 12; i++) {
-    n = parseInt(numbers[i])
-    if (i % 2 === 0) {
-      checkSum += n
-    } else {
-      checkSum += ((n * 2) % 9) + Math.floor(n / 9) * 9
-    }
-  }
+export const isPersonIdValid = (id: string): boolean => isPersonnummerValid(id) || isSamordningsnummerValid(id)
 
-  return checkSum % 10 === 0
-}
-
-export const formatPersonId = (id: string) => {
+export const formatPersonId = (id: string): string => {
   let cleanId = id.replace(/\D/g, '')
   if (cleanId.length > 8) {
     cleanId = cleanId.slice(0, 8) + '-' + cleanId.slice(8, cleanId.length)
   }
   return cleanId
 }
+
+const isPersonnummerValid = (value: string): boolean => {
+  const PERSONNUMMER_REGEXP = /^(\d{2})(\d{2})(\d{2})([0-3]\d)([-+]?)(\d{4})$/
+  const parts = PERSONNUMMER_REGEXP.exec(value)
+
+  if (!parts) return false
+
+  const parsedDate = parseISO(parts[1] + parts[2] + '-' + parts[3] + '-' + parts[4])
+
+  if (!isValid(parsedDate) || isFuture(parsedDate) || !isDateInValidRange(parsedDate)) {
+    return false
+  }
+
+  return isCheckDigitValid(parts[2] + parts[3] + parts[4] + parts[6])
```

Comment:
Since the parts are used in several places could the algorithm be easier to read if we saved them in variables with explainable names? Right now it's hard to understand what is in each part. Same goes for coordination number